ledfx/effects/gifplayer.py

```diff
@@ -0,0 +1,149 @@
+                default=False,
+            ): bool,
+            vol.Optional(
+                "url source", description="Load image from", default=""
+            ): str,
+            vol.Optional(
+                "bounce",
+                description="Bounce the GIF instead of looping",
+                default=False,
+            ): bool,
+            vol.Optional(
+                "GIF FPS", description="How fast to play the gif", default=10
+            ): vol.All(vol.Coerce(int), vol.Range(min=1, max=60)),
+        }
+    )
+
+    def __init__(self, ledfx, config):
+        super().__init__(ledfx, config)
+
+    def config_updated(self, config):
+        super().config_updated(config)
+        self.gif_fps = self._config["GIF FPS"]
+        self.bounce = self._config["bounce"]
+        self.resize = (
+            Image.BILINEAR if self._config["bilinear"] else Image.NEAREST
+        )
+        self.frames = []
+        self.current_frame = 0
+        self.start_time = time.time()
+        self.last_frame_time = self.start_time
+        self.init = True
+
+    def audio_data_updated(self, data):
+        # Non-reactive by design - do nothing
+        return
+
+    def open_image(self, image_path):
+        try:
+            if image_path.startswith("http://") or image_path.startswith(
+                "https://" or image_path.startswith("file://")
+            ):
+                with urllib.request.urlopen(image_path) as url:
+                    return Image.open(url)
+            else:
+                return Image.open(image_path)  # Directly open for local files
+        except Exception as e:
+            _LOGGER.error("Failed to open image, using default GIF: %s", e)
+            return Image.open(self.DEFAULT_GIF_PATH)
+
+    def do_once(self):
+        super().do_once()
+
+        url_path = self._config["url source"]
+
+        if url_path == "":
+            self.gif = self.open_image(self.DEFAULT_GIF_PATH)
+        else:
+            self.gif = self.open_image(url_path)
+        self.process_gif()
+        # Seed the frame data with the first frame
+        self.frame_data = self.frames[0]
+        self.gif_frame_duration = 1 / self.gif_fps
+        self.init = False
+
+    def draw(self):
+        if self.init:
+            self.do_once()
+        current_time = time.time()
+        self.step_gif_if_time_elapsed(current_time)
+        self.matrix.paste(self.frame_data)
+
+    def step_gif_if_time_elapsed(self, current_time):
+        """
+        Steps to the next frame of the GIF if the specified time has elapsed.
+
+        Args:
+            current_time (float): The current time in seconds.
+
+        Returns:
+            None
+        """
+        if current_time - self.last_frame_time >= self.gif_frame_duration:
+            if self.bounce:
+                if self.current_frame == (len(self.frames) - 1):
```

Comment:
use >= just in case

Reply:
I can't really see a case where we go out of bounds here - if we are truly somehow at a frame thats greater than the current frame, we've already exploded since we have already tried to to assign the frame_data of a non-existent frame in line 114.

The intention is clearer using explicit comparison.